tern/analyze/common.py

```diff
@@ -452,3 +454,100 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_folder(path_to_folder):
+    '''Given a path to git folder, return the name and sha of repository.
+    This function looks into .git folder.'''
+    comment_line = ''
+    if os.path.isdir(path_to_folder):
+        # find .git folder
+        dot_git_folder_path = os.path.join(path_to_folder, '.git/')
+        if os.path.exists(dot_git_folder_path):
+            # find sha information
+            sha_info = ''
+            with open(dot_git_folder_path + 'HEAD', 'r') as head_file:
+                # .git/HEAD contains current head ref
+                ref_info = head_file.readline()
+                ref_relative_path = ref_info.split(' ', 1)[1].strip('\n')
+                sha_path = dot_git_folder_path + ref_relative_path
+                # get sha info from ref
+                with open(sha_path, 'r') as sha_file:
+                    sha_info = sha_file.readline().strip('\n')
```

Comment:
You can get the sha info in one line using `git rev-parse HEAD`. Take a look at `get_git_rev_or_version()` function in  tern/tern/utils/general.py for an example on how to do this. You may also want to break this into a mini-function called something like `get_git_sha(path)` for clarity.